### PR TITLE
LOG4J2-3121 config modified at run-time may trigger incomplete MBean re-initialization due to InstanceAlreadyExistsException

### PR DESCRIPTION
**Pull request by sman-81**
…e MBean re-initialization due to InstanceAlreadyExistsException

### CODE REVIEW

File: log4j-core/src/main/java/org/apache/logging/log4j/core/jmx/Server.java [hunk 1]
```diff
@@ -389,7 +389,15 @@ private static void registerAppenders(final LoggerContext ctx, final MBeanServer
 
     private static void register(final MBeanServer mbs, final Object mbean, final ObjectName objectName)
             throws InstanceAlreadyExistsException, MBeanRegistrationException, NotCompliantMBeanException {
+        if (mbs.isRegistered(objectName)) {
+            try {
+                mbs.unregisterMBean(objectName);
+            } catch (MBeanRegistrationException | InstanceNotFoundException _ex) {
+                LOGGER.trace("Failed to unregister MBean {}", objectName);
```
Comment (garydgregory) [inline log4j-core/src/main/java/org/apache/logging/log4j/core/jmx/Server.java:396]: No underscores please, _ex -> ex

**Review by garydgregory — CHANGES_REQUESTED** [review]
See comment.

File: log4j-core/src/main/java/org/apache/logging/log4j/core/jmx/Server.java (hunk 1 above)
Comment (sman-81) [reply log4j-core/src/main/java/org/apache/logging/log4j/core/jmx/Server.java:396]: Fixed

**Comment by vy** [comment]
@sman-81, given this addresses a bug, I'd appreciate a change in `changes.xml` too. (You can use `dev="ggregory"`.)

**Comment by sman-81** [comment]
> @sman-81, given this addresses a bug, I'd appreciate a change in `changes.xml` too. (You can use `dev="ggregory"`.)
Done. I'm still new to the process 😎
I'm not sure why there's a merge conflict on `changes.xml` now, though.

**Comment by vy** [comment]
@garydgregory, I have resolved the merge conflict. Would you mind squashing & merging this, please? (Cherry-picking into `master` would be great.)

**Comment by garydgregory** [comment]
Hi @vy 
I brought this PR into the `release-2.x` branch but I cannot get Eclipse to build anything in master. I'll put some time in to investigate but it might be an aspect of what @rgoers wrote up on the ML.

**Pull request merged**